pkg/auth/authmap_gc.go

```diff
@@ -405,30 +408,46 @@ func (r *authMapGarbageCollector) cleanupExpiredEntries(_ context.Context) error
 
 func (r *authMapGarbageCollector) subscribeToEndpointEvents(endpointManager endpointmanager.EndpointManager) {
 	endpointManager.Subscribe(r)
```

Comment:
Could move the `Subscribe` call into the below critical section unless I'm missing a reason not to. Subscribing before it allows events to come in prior to acquiring the snapshot, and that feels a bit racy (though I can't see an actual bug).